### ISSUE TITLE
Flag identical boolean expressions in an if statement

### ISSUE THREAD

**Issue by filiph** [dart-lang/linter#173]
```dart
if (View.StereoPass == eSSP_LEFT_EYE ||
    View.StereoPass == eSSP_LEFT_EYE) {
  ...
}
```

Common copy-paste bug.

**Comment by alexeieleusis** [comment]
This might be linted now that https://github.com/dart-lang/linter/pull/229 is merged. I might add a test to confirm @pq can you assign this to me?

**Comment by filiph** [comment]
Is this in?

**Comment by alexeieleusis** [comment]
It is linted only if the boolean operator is `&&`.

**Comment by pq** [comment]
> It is linted only if the boolean operator is &&.

@alexeieleusis : are you working on extending it?

Thanks!

**Comment by alexeieleusis** [comment]
No. But I am surprised that is not linted, I just reviewed tests and we have a few that lint expressions where the operator is `||`, so looks like a bug to me. We can leave this open and I will look into it.

Just to set the right expectations, I am a bit low on cycles for this now, but will eventually get to it.

**Comment by pq** [comment]
NP.  Thanks!
